src/common/logging.hpp

```diff
@@ -160,6 +161,24 @@ void otbrLogDeinit(void);
                 ##__VA_ARGS__, otbrErrorString(_err));                                                    \
     } while (0)
 
+/**
+ * This macro log an action result according to @p aError, where aError is an non-OTBR error code.
+ *
+ * If @p aError is zero, the log level will be OTBR_LOG_INFO, otherwise OTBR_LOG_WARNING.
+ *
+ * @param[in] aError   The action result.
+ * @param[in] aFormat  Format string as in printf.
+ * @param[in] ...      Arguments for the format specification.
+ *
+ */
+#define otbrLogResultCode(aError, aFormat, ...)                                                           \
+    do                                                                                                    \
+    {                                                                                                     \
+        int32_t _err = static_cast<int32_t>(aError);                                                      \
```

Comment:
```suggestion
        int _err = static_cast<int>(aError);                                                      \
```

why not use `int`?

Reply:
I'm afraid that `int` is not big enough, `int` has to be at least 32bit to make sure it's not overflowing. For example, `kDNSServiceErr_BadParam = -65540`.